### PR TITLE
Update to v1.8.0-1.rc

### PR DESCRIPTION
**Pull request by jciupis**

### CODE REVIEW

File: release_notes/nrf_802154_1.8.0.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# nRF IEEE 802.15.4 radio driver 1.8.0 release notes
+
+Release date: August 11, 2020
```
[the remark this answers is not in the document]
Comment (jciupis) [reply release_notes/nrf_802154_1.8.0.md]: Removed the line entirely.

File: release_notes/nrf_802154_1.8.0.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Depending on the application's target platform, either one of those files should be added to the build system, or yet another implementation of the API should be provided.
+
+Moreover, only a single implementation of API defined in src/platform/nrf_802154_lp_timer.h is provided:
+* src/platform/lp_timer/nrf_802154_lp_timer.c
+
+Any other implementation used by the build system should be replaced with the above file.
+
+## New features
+* Added interrupt handling abstraction layer for easier integration with operating systems. See [PR #358](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/358)
```
[the remark this answers is not in the document]
Comment (jciupis) [reply release_notes/nrf_802154_1.8.0.md]: Done.

File: release_notes/nrf_802154_1.8.0.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
```
[the remark this answers is not in the document]
Comment (jciupis) [reply release_notes/nrf_802154_1.8.0.md:23]: Cool suggestion! Done

**Comment by hubertmis** [comment]
I think that @czeslawmakarski 's review is needed and we'll merge

**Pull request merged**